src/utils/error-handler.ts

```diff
@@ -0,0 +1,58 @@
+import { Context } from 'hono';
+import { DBError, ValidationError } from './custom-error';
+import { errorCode } from './utils';
+import { formatZodError } from './error-formater';
+import { HTTPException } from 'hono/http-exception';
+
+export function DBErrorHandler(err: Error, c: Context) {
+  if (err instanceof DBError) {
+    switch (err.code) {
+      case errorCode.CONNECTION: {
+        c.status(502);
+        break;
+      }
+      case errorCode.INTERNAL_SERVER_ERROR: {
+        c.status(500);
+        break;
+      }
+      case errorCode.NOT_FOUND: {
+        c.status(404);
+        break;
+      }
+    }
+  } else if (err) {
+    formatZodError(err, c);
+  }
+
+  return c.json(err.message);
+}
+
+export function ValidationErrorHandler(err: Error, c: Context) {
+  if (err instanceof ValidationError) {
+    switch (err.code) {
+      case errorCode.INVALID: {
+        c.status(400);
+        break;
+      }
+      case errorCode.NOT_FOUND: {
+        c.status(404);
+        break;
+      }
+      case errorCode.UNAUTHORIZED: {
+        c.status(401);
+        break;
+      }
+    }
+  } else if (err instanceof HTTPException) {
+    if (!err.getResponse().statusText) {
+      c.status(401);
+      return c.json('Missing authorization... You need to sign in and get access token');
```

Comment:
Please remove this part of the sentence “and get access token”. It’s not secure to share what our authorisation method is